core/parser/src/main/antlr4/imports/JPQL_lexer.g4

```diff
@@ -209,7 +209,9 @@ fragment NOT_FRAG : [Nn][Oo][Tt];
 
 fragment Date_string : DIGIT DIGIT DIGIT DIGIT '-' DIGIT DIGIT '-' DIGIT DIGIT;
 
-fragment Time_string : DIGIT DIGIT? ':' DIGIT DIGIT ':' DIGIT DIGIT ('.' DIGIT*)?;
+fragment Time_string : DIGIT DIGIT? ':' DIGIT DIGIT ':' DIGIT DIGIT;
+
+fragment Nanos_string : ('.' DIGIT*)?;
```

Comment:
A normal JDBC escaped time has no nanos, so this was wrong before. Only the timestamp has nanos.